analyzers/its/expected/BlazorSample/BlazorSample--net7.0-S1144.json

```diff
@@ -0,0 +1,30 @@
+{
+"issues":  [
```

Comment:
- Only SomeMethod is detected, not the other two unused private methods => Rule improvement issue? 
- No issues in razor files due to the limitation of the SL context => OK

Reply:
I think so. But the rule is quite complex so maybe we should look into it a bit to see whether those cases are excluded on purpose and why (maybe the fact that there is an attribute?)

Reply:
Concerning the FN of the local function: we already have an [issue in the backlog](https://github.com/SonarSource/sonar-dotnet/issues/6699), opened earlier this year. So, specifically regarding local functions, there is nothing new to add to the backlog.

Concerning  the FNs of the private methods with attributes: indeed, removing the attribute makes the issue appear on the `private` method:
```cs
public class S6798_CSharpOnly
{
    private void JSInvokablePrivateMethodInPartialRazorComponent() { } // Removing JSInvokable makes a S1144 issue appear here
    // ...
}
```

I don't see, however, why any attribute should alter the behavior of the rule in such a way.
Moreover, the only example with an attribute that I can see from the tests of the rule is the following:
```cs
[My] 
private class Class1 { }
```
This scenario doesn't cover `private` methods with attributes. So I am going to add a new issue to the backlog, and a reproducer to the tests of S1144.

Reply:
Repro: https://github.com/SonarSource/sonar-dotnet/pull/8349
Issue: https://github.com/SonarSource/sonar-dotnet/issues/8348